### PR TITLE
Electron 17 - Crash Reporter Implementation

### PR DESCRIPTION
**Pull request by VishwasShashidhar**
- Supports generating crash reports from main and renderer processes
- Supports opening the crashes directory from the "View" menu
- Currently, the company name is hard coded
- Mini dumps are generated in the temp directory of the OS

### CODE REVIEW

**Comment by VishwasShashidhar** [comment]
@VikasShashidhar @lneir Please review.

File: js/windowMgr.js [hunk 1]
```diff
@@ -180,6 +181,24 @@ function doCreateMainWindow(initialUrl, initialBounds) {
...
+            title: 'Renderer Process Crashed',
+            message: 'Oops! Looks like we have had a crash. Please reload or close this window.',
+            buttons: ['Reload', 'Close']
+        };
+
+        electron.dialog.showMessageBox(options, function (index) {
+            if (index === 0) {
+                mainWindow.reload();
+            }
+            else mainWindow.close();
```
Comment (lneir) [inline js/windowMgr.js]: missing curly braces.

File: js/windowMgr.js [hunk 2]
```diff
@@ -180,6 +181,24 @@ function doCreateMainWindow(initialUrl, initialBounds) {
         loadErrors.showLoadFailure(mainWindow, validatedURL, errorDesc, errorCode, retry, false);
     });
 
+    // In case a renderer process crashes, provide an
+    // option for the user to either reload or close the window
+    mainWindow.webContents.on('crashed', function () {
+        const options = {
+            type: 'error',
+            title: 'Renderer Process Crashed',
+            message: 'Oops! Looks like we have had a crash. Please reload or close this window.',
```
Comment (lneir) [inline js/windowMgr.js]: i prefer: oh snap :-)  just kidding.

File: js/preload/preloadMain.js [hunk 3]
```diff
@@ -110,6 +111,13 @@ function createAPI() {
         ScreenSnippet: remote.require('./screenSnippet/index.js').ScreenSnippet,
 
         /**
+         * Provides API to crash the renderer process that calls this function
+         */
+        crashRendererProcess: function () {
```
Comment (lneir) [inline js/preload/preloadMain.js]: is there away to only allow this to be run for dev builds?

File: js/windowMgr.js (hunk 2 above)
Comment (VishwasShashidhar) [reply js/windowMgr.js]: he he :-)

File: js/preload/preloadMain.js (hunk 3 above)
Comment (VishwasShashidhar) [reply js/preload/preloadMain.js]: we can try with `process.env.ELECTRON_DEV` and see if that works on preload.

File: js/windowMgr.js (hunk 1 above)
Comment (VishwasShashidhar) [reply js/windowMgr.js]: noted. will fix it.

**Comment by VishwasShashidhar** [comment]
@lneir @VikasShashidhar Made corrections as per the suggestions above. Please review.

File: js/main.js [hunk 4]
```diff
@@ -31,6 +32,8 @@ require('./memoryMonitor.js');
 
 const windowMgr = require('./windowMgr.js');
 
+crashReporter.start({companyName: 'Symphony', submitURL: 'http://localhost:3000', uploadToServer: false, extra: {'process': 'main'}});
```
Comment (lneir) [inline js/main.js]: why the http://localhost:3000???

Comment (VishwasShashidhar) [reply js/main.js]: It's a default value. We can change it when we have the server up. Unfortunately, submitURL is a mandatory attribute.

Comment (lneir) [reply js/main.js]: sounds good.

**Comment by lneir** [comment]
lgtm.

looks like PR needs rebasing as commit history above has other commits.

**Comment by VishwasShashidhar** [comment]
@lneir Just made a small change of opening the "completed" sub-directory when an user clicks on the "Open Crashes Directory" from the "View" menu.

Please see if that's fine. I'll rebase the branch and push a commit.

**Comment by VikasShashidhar** [comment]
Alright, I am merging this.

**Pull request merged**